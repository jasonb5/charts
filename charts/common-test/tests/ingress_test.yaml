suite: test ingress
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: deployment

      ingress:
        className: traefik
        hosts:
          default:
            - name: default
              path: /index
              pathType: ImplementationSpecific
            - name: https
              path: /version
          domain.io:
            - name: https
              path: /login
              pathType: Prefix
        tls: tls-certs

      service:
        ports:
          default:
            port: 80
          https:
            port: 443
    asserts:
      - matchSnapshot: {}
        documentIndex: 2
      - hasDocuments:
          count: 3
  - it: should pass
    set:
      workload: deployment

      ingress:
        hosts:
          default:
            - name: default
          domain.io:
            - name: https

      service:
        ports:
          default:
            port: 80
          https:
            port: 443
    asserts:
      - matchSnapshot: {}
        documentIndex: 2
      - hasDocuments:
          count: 3
