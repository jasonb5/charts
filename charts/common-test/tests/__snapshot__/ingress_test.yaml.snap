should overwrite defaults:
  1: |
    apiVersion: v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      ingressClassName: traefik
      rules:
        - http:
            paths:
              - backend:
                  name: default
                  port:
                    number: 80
                path: /index
                pathType: ImplementationSpecific
              - backend:
                  name: https
                  port:
                    number: 443
                path: /version
        - host: domain.io
          http:
            paths:
              - backend:
                  name: https
                  port:
                    number: 443
                path: /login
                pathType: Prefix
      tls:
        - hosts:
            - domain.io
          secretName: RELEASE-NAME-common-test-tls-certs
should pass:
  1: |
    apiVersion: v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      rules:
        - http:
            paths:
              - backend:
                  name: default
                  port:
                    number: 80
                path: /
        - host: domain.io
          http:
            paths:
              - backend:
                  name: https
                  port:
                    number: 443
                path: /
